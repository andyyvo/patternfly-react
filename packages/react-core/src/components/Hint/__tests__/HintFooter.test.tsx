import React from "react";
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { HintFooter } from "../HintFooter";

describe('HintFooter', () => {
  it('renders without children', () => {
    render(<HintFooter data-testid="HintFooter"></HintFooter>);
    
    expect(screen.getByTestId('HintFooter')).toBeVisible();
  });

  it('renders children', () => {
    render(<HintFooter>{<button>Test Me</button>}</HintFooter>);

    expect(screen.getByRole('button', { name: 'Test Me' })).toBeVisible();
  });

  it('renders with class pf-c-hint__footer', () => {
    render(<HintFooter>Hint Body Test</HintFooter>);

    const body = screen.getByText('Hint Body Test');

    expect(body).toHaveClass('pf-c-hint__footer');
  });

  it('renders with custom class names provided via prop', () => {
    render(<HintFooter className="custom-classname">Hint Body Test</HintFooter>);

    const body = screen.getByText('Hint Body Test');

    expect(body).toHaveClass('custom-classname');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<HintFooter>{<button>Test</button>}</HintFooter>);

    expect(asFragment()).toMatchSnapshot();
  });
});