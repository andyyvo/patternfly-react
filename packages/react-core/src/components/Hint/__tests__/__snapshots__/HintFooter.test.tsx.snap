// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintFooter matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-c-hint__footer"
  >
    <button>
      Test
    </button>
  </div>
</DocumentFragment>
`;
